src/Query/DataReaderInterface.php

```diff
@@ -48,6 +49,14 @@ public function key(): int|string|null;
      */
     public function current(): array|object|false;
 
+    /**
+     * Sets the columns for type casting the query results.
+     * Do not use this method if you want to get the raw data from the query.
+     *
+     * @param ColumnInterface[] $columns
+     */
+    public function columns(array $columns): static;
```

Comment:
```suggestion
    public function typecastColumns(array $columns): static;
```

`columns` is confused. As if only these columns should be returned.

Reply:
Maybe `resultColumns()`